lib/ansible/modules/network/ios/ios_pmtud.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import print_function
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: ios_pmtud
+short_description: Perform path MTU discovery on Cisco IOS devices.
+description:
+    - Perform path MTU discovery on Cisco IOS devices.
+author:
+    - Martin Komon (@mkomon)
+version_added: '2.5'
+extends_documentation_fragment: ios
+options:
+    dest:
+        description:
+            - IP to ping, should respond to 1500 B packets from Internet
+        required: true
+    vrf:
+        description:
+            - VRF name
+        required: false
+    source:
+        description:
+            - Source interface name or IP address
+        required: false
+    max_size:
+        description:
+            - Start and max size for path MTU discovery.
+        default: 1500
+        required: false
+    max_range:
+        description:
+            - Max range of path MTU discovery. Must be 2^n.
```

Comment:
Would be be worth adding some code to ensure that `max_range` is `2^n`?

Reply:
It certainly would, thanks for input!
I've added a check and revert to default value when check fails.

Reply:
@gundalow How do you like the fix? Please let me know if it needs more work, thank you.